internal/cmd/skupper/listener/listener_create.go

```diff
@@ -1,47 +1,188 @@
+	} else if len(args) > 2 {
+		validationErrors = append(validationErrors, fmt.Errorf("only two arguments are allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener name must not be empty"))
+	} else if args[1] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener port must not be empty"))
+	} else {
+		cmd.name = args[0]
+		ok, err := resourceStringValidator.Evaluate(cmd.name)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener name is not valid: %s", err))
+		}
+
+		cmd.port, err = strconv.Atoi(args[1])
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+		ok, err = numberValidator.Evaluate(cmd.port)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+	}
+
+	// Validate if there is already a listener with this name in the namespace
+	if cmd.name != "" {
+		listener, err := cmd.client.Listeners(cmd.namespace).Get(context.TODO(), cmd.name, metav1.GetOptions{})
+		if listener != nil && !errors.IsNotFound(err) && listener.Status.StatusMessage == "Ok" {
+			validationErrors = append(validationErrors, fmt.Errorf("there is already a listener %s created for namespace %s", cmd.name, cmd.namespace))
+		}
+	}
+
+	// Validate flags
+	//TBD what characters are not allowed
+	if cmd.flags.routingKey != "" {
+		ok, err := resourceStringValidator.Evaluate(cmd.flags.routingKey)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("routing key is not valid: %s", err))
+		}
+	}
+	if cmd.flags.host != "" {
+		ok, err := resourceStringValidator.Evaluate(cmd.flags.host)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("host name is not valid: %s", err))
+		}
+	}
+	if cmd.flags.tlsSecret != "" {
+		ok, err := tlsStringValidator.Evaluate(cmd.flags.tlsSecret)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("tls-secret is not valid: %s", err))
+		}
+	}
+	if cmd.flags.listenerType != "" {
+		ok, err := listenerTypeValidator.Evaluate(cmd.flags.listenerType)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener type is not valid: %s", err))
+		}
+	}
+	return validationErrors
+}
+
+func (cmd *CmdListenerCreate) Run() error {
+
+	listener := v1alpha1.Listener{
+		ObjectMeta: metav1.ObjectMeta{Name: cmd.name},
+		Spec: v1alpha1.ListenerSpec{
+			Host:           cmd.flags.host,
+			Port:           cmd.port,
+			RoutingKey:     cmd.flags.routingKey,
+			TlsCredentials: cmd.flags.tlsSecret,
+			Type:           cmd.flags.listenerType,
+		},
+	}
+
+	_, err := cmd.client.Listeners(cmd.namespace).Create(context.TODO(), &listener, metav1.CreateOptions{})
+	return err
+}
+
+func (cmd *CmdListenerCreate) WaitUntilReady() error {
+	err := utils.NewSpinner("Waiting for Listener...", 5, func() error {
```

Comment:
Instead of "Waiting for Listener..." -> "Waiting for status..."

(I missed this documentation change in my site create command implementation, I will update it as well there)

Reply:
done